src/Domain/Insights/SniffDecorator.php

```diff
@@ -152,4 +121,36 @@ public function addDetails(Details $details): void
     {
         $this->errors[] = $details;
     }
+
+    private function skipFilesFromIgnoreFiles(InsightFile $file): bool
+    {
+        $path = $file->getFileInfo()->getRealPath();
+
+        if ($path === false) {
+            return false;
+        }
+
+        foreach ($this->getIgnoredFilesPath() as $ignoredFilePath) {
+            if (self::pathsAreEqual($this->dir . DIRECTORY_SEPARATOR . $ignoredFilePath, $path)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Contains the setting for all files which the sniff should ignore.
+     *
+     * @return array<int, string>
+     */
+    private function getIgnoredFilesPath(): array
+    {
+        return $this->sniff->exclude ?? [];
+    }
+
+    private static function pathsAreEqual(string $pathA, string $pathB): bool
```

Comment:
This part is now duplicated in both sniff decorator and fixer decorator. Could you refactor it to some kind of helper class? :)

Reply:
Also because we later on has to add logic to it, so it can take regexes and folders and so on.

Reply:
I create a PathComparator Helper to handle this ;)